configure.py

```diff
@@ -2982,6 +2998,37 @@ def canonicalize_options(options, info_os, info_arch):
     if options.cpu != options.arch:
         logging.info('Canonicalized CPU target %s to %s', options.cpu, options.arch)
 
+    # select and sanity check build targets
+    def canonicalize_build_targets(options):
+        build_targets = read_build_targets(options)
+
+        # --build-targets was not provided: build default targets
+        if build_targets == None:
+            options.build_targets = "cli,tests"
+            return
```

Comment:
This ensures legacy behaviour by selecting the targets to `cli,tests`. Furthermore, it must skip the sanity checks below, to fully leave control over static and shared libraries to the legacy switches `--(enable/disable)-(shared/static)-library)`.